disttask/framework/storage/task_table.go

```diff
@@ -706,3 +708,30 @@ func (stm *TaskManager) GetSubtasksByStep(taskID, step int64) ([]*proto.Subtask,
 	}
 	return subtasks, nil
 }
+
+// TransferSubTasks2History move all the finished subTask to tidb_background_subtask_history by taskID
+func (stm *TaskManager) TransferSubTasks2History(taskID int64) error {
+	return stm.WithNewTxn(stm.ctx, func(se sessionctx.Context) error {
+		_, err := ExecSQL(stm.ctx, se, "insert into mysql.tidb_background_subtask_history select * from mysql.tidb_background_subtask where task_key = %?", taskID)
+		if err != nil {
+			return err
+		}
+
+		// delete taskID subtask
+		_, err = ExecSQL(stm.ctx, se, "delete from mysql.tidb_background_subtask where task_key = %?", taskID)
+		return err
+	})
+}
+
+// GC deletes the history subtask which is older than the given days.
+func (stm *TaskManager) GC() error {
+	subtaskHistoryKeepDays := defaultSubtaskKeepDays
+	failpoint.Inject("SubtaskHistoryKeepDays", func() {
+		subtaskHistoryKeepDays = 0
+	})
+	_, err := stm.executeSQLWithNewSession(
+		stm.ctx,
+		fmt.Sprintf("DELETE FROM mysql.tidb_background_subtask_history WHERE state_update_time < UNIX_TIMESTAMP() - %d * 24 * 60 * 60;", subtaskHistoryKeepDays),
```

Comment:
Could you add a test that some records are less than 14 days?

Reply:
Do you worry about the test of `subtaskHistoryKeepDays=0` is not enough?🤔

Reply:
I mean delete part of the records but not all of them.

Reply:
added it